app/routes/index.js

```diff
@@ -10,56 +10,68 @@ module.exports = function(app, dbURL) {
 		res.render('index');
 	});
 
-	// Matches paramter in short url format
+	// Matches short url
 	app.get(/^\/[0-9]{7}$/, function(req,res) {
 		var paramUrl = req.url.substr(1);
 		winston.log('info', 'Sending URL parameter to short-URL handler');
 		// find and return document with matching short-URL value
-		shortURLHandler(paramUrl, dbURL, function(response) {
-			if (response) {
+		shortURLHandler(paramUrl, dbURL).then(function(result) {
+			if (result) {
 				// this redirects user to the original_url value stored in the database
 				// with status 302 for found
-				res.redirect(response.original_url);
-				winston.log('info', 'User redirected to: ' + response.original_url + '\n');
+				res.redirect(result.original_url);
+				winston.log('info', 'User redirected to: ' + result.original_url + '\n');
 			} else {
 				winston.log('info', 'Short URL does not exist\n');
-				res.status(500).json({error: 'Short URL does not exist'});
+				res.status(404).json({
+					error: 'Short URL does not exist'
+				});
 			}
+		}).catch(function(reason) {
+			winston.log('error', reason);
+			res.status(500).json({
+				error: 'Internal Server Error'
+			});
 		});
 	});
 
+	// Matches original url
 	app.get('/*', function(req,res) {
 		var paramUrl = req.url.substr(1);
 		winston.log('info', 'URL Parameter: ' + paramUrl);
 
-		// if parameter was an original url
+		// if parameter is a valid uri
 		if (validUrl.isUri(paramUrl)) {
-			// normalizes url to prevent duplication under different names e.g. https://www.google.com // https://google.com
+			// normalizes url to prevent duplication under different names
+			// e.g. https://www.google.com // https://google.com
 			paramUrl = normalizeUrl(paramUrl);
 			winston.log('info', 'Sending normalized URL parameter to URL handler as: ' + paramUrl);
-			originalURLHandler(paramUrl, dbURL, function(type, response) {
-				// working with response from findURL
-				if (type == 'URLExisted') {
+			originalURLHandler(paramUrl, dbURL).then(function(result) {
+				// result.ops is response from an MongoDB Insert command
+				// result.original_url is response from MongoDB findOne command
+				// Either way, we want to respond with the result.
+				if (result.ops) {
 					res.status(200).json({
-						original_url: response.original_url,
-						short_url: response.short_url
+						original_url: result.ops[0].original_url,
```

Comment:
I see your point here. If `result.ops` and/or `result.ops[0]` return null or undefined, this would imply there was an error connecting to the database, or that the data was stored incorrectly. If this occurs, the catch function will return an internal server error, and my app will continue to run (Promises make this part of the problem so easy). 

I'm not sure what to do if the result from the database is different from what I expect, other than log and return the error.